Make ProtectionProfile methods public

sirzooro opened the pull request:
#### Description
Make ProtectionProfile methods public. This allows to get values like key length and use them with key management protocols like MIKEY, instead of creating own constants for this.

#### Reference issue
Fixes #258

at-wat commented on srtp_cipher.go:
```diff
@@ -10,11 +10,11 @@ import "github.com/pion/rtp"
-	rtpAuthTagLen() (int, error)
-	rtcpAuthTagLen() (int, error)
+	RtpAuthTagLen() (int, error)
+	RtcpAuthTagLen() (int, error)
 	// aeadAuthTagLen returns AEAD auth key length of the cipher.
 	// See the note below.
-	aeadAuthTagLen() (int, error)
+	AeadAuthTagLen() (int, error)
```
Acronyms are all capitals in Go's naming convention. So `RTPAuth*`, `RTCPAuth*`, `AEADAuth*` would be preferred

sirzooro replied:
fixed

at-wat reviewed (commented):
Added comment on naming.
The idea looks good to me!

sirzooro commented:
I have changed names as requested and added comments to make linter happy.

Please take a look on test pipeline, it failed with following error:
[2024-02-24T04:11:55.282Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}

Sean-Der commented:
@sirzooro Great!

Can you squash those commits into one. Use your description on GitHub as the commit message.

I added you to the GitHub org so the tests should run (without our intervention). When it goes green we can merge :)

sirzooro commented:
Done :)

Sean-Der commented:
Perfect! Thank you so much @sirzooro 

Feel free to push a tag if you need one. `git tag v3.0.1 && git push --tags` is all the process you need :)

at-wat reviewed (approved):
LGTM
